reactome_analysis_api/reactome_analysis_api/__main__.py

```diff
@@ -80,6 +83,133 @@ def custom_abort(code: int, message: str):
 def show_mainpage():
     return redirect("/0.1/ui")
 
+
+@app.route("/uploadRibo", methods=["POST"])
+def upload_ribo():
+    store_file = request.args.get('store', 'true').lower() == "true"
+
+    if 'fileRibo' not in request.files or 'fileRNA' not in request.files:
+        return custom_abort(400, "Incorrect number of uploaded files. Function requires two files.")
+
+    if "fileRibo" not in request.files or "fileRNA" not in request.files:
+        return custom_abort(400, "File must be uploaded as 'file' in the form.")
+
+    # get the uploaded file
+    file_rna = request.files['fileRNA']
+    user_file_rna = file_rna.filename
+
+    file_ribo = request.files['fileRibo']
+    user_file_ribo = file_ribo.filename
+
+    # initialize the return object
+    return_object = {"sample_names": None, "top_identifiers": list(), "n_lines": None}
+    return_lines = list()
+    n_samples = -1
+
+    # read the file
+    try:
+        all_lines_rna = [line.decode("UTF-8") for line in file_rna.readlines()]
+        all_lines_ribo = [line.decode("UTF-8") for line in file_ribo.readlines()]
+    except Exception as e:
+        # get the extension
+        (name, rna_extension) = os.path.splitext(user_file_rna)
+        (name, ribo_extension) = os.path.splitext(user_file_ribo)
+
+        if rna_extension == ".xlsx" or ribo_extension == ".xlsx":
+            UPLOAD_ERRORS.labels(extension=rna_extension + ribo_extension).inc()
```

Comment:
changed to xslx but I would agree with you since the user base is not that big and we do not perform any analytics on basis of the data the counter is not necessar.